### PR TITLE
Proxy bblfsh

### PR DESCRIPTION
**Pull request by smacker**
Fix: #339

### CODE REVIEW

File: cmd/lookoutd/common.go [hunk 1]
```diff
@@ -375,8 +379,10 @@ func (c *queueConsumerCommand) startServer(srv *lookout.DataServerHandler) error
 	go func() {
 		if err := grpcSrv.Serve(lis); err != nil {
 			log.Errorf(err, "data server failed")
+			os.Exit(1)
```
Comment (carlosms) [inline cmd/lookoutd/common.go]: Why is this needed now? Are there any changes that require it, or was it missing from before?

Comment (smacker) [reply cmd/lookoutd/common.go]: it was missed before I just noticed and fixed it while doing my task.

Currently, if data-server fails we log it but don't exit.
Worker will continue processing events but all of them will fail because analyzers won't be able to connect to data-server to get changes.

**Comment by smacker** [comment]
all tests are passing successfully locally but fail in travis (with errors like `listen tcp 0.0.0.0:8090: bind: address already in use`). I'm investigating.

File: cmd/lookoutd/common.go (hunk 1 above)
Comment (carlosms) [reply cmd/lookoutd/common.go]: How did you stumble with the problem? Can this happen only on initialization, like an already used port, or at any time?

Comment (smacker) [reply cmd/lookoutd/common.go]: I just noticed it while looking at the code around.
I didn't dig too deep in grpc implementation to answer your question. But for http server it can return error in 2 cases on initialization and when done channel is closed. (happens when you call `server.Shutdown()` or `server.Close()`) Most probably grpc works similar.

Comment (carlosms) [reply cmd/lookoutd/common.go]: What I was thinking is that maybe this method should be blocking, removing `go func` and directly do `return grpcSrv.Serve`. Then on top handle it like here https://github.com/src-d/lookout/blob/master/cmd/lookoutd/serve.go#L96. WDYT? is it worth to make it consistent with other top level errors, so we have one clear exit point?

Comment (smacker) [reply cmd/lookoutd/common.go]: We use this function not only in `serve.go` but in worker too. It doesn't use chan/switch pattern. SDK uses a similar pattern but instead of making the methods blocking it accepts chan as an argument. 

Though I like your idea to make it consistent, I'm not sure which pattern we should use. Do you mind if we move it to another issue then?

Comment (carlosms) [reply cmd/lookoutd/common.go]: ok, let's keep it as it is here, and I'll open an issue to look into it later, it's a minor thing and not really urgent.

Comment (carlosms) [reply cmd/lookoutd/common.go]: Here: https://github.com/src-d/lookout/issues/362

File: cmd/lookout-sdk/event.go [hunk 2]
```diff
@@ -129,7 +129,12 @@ func (c *EventCommand) makeDataServerHandler() (*lookout.DataServerHandler, erro
 
 func (c *EventCommand) bindDataServer(srv *lookout.DataServerHandler, serveResult chan error) (*grpc.Server, error) {
 	log.Infof("starting a DataServer at %s", c.DataServer)
-	grpcSrv := grpchelper.NewServer()
+	bblfshGrpcAddr, err := grpchelper.ToGoGrpcAddress(c.Bblfshd)
```
Comment (dpordomingo) [inline cmd/lookout-sdk/event.go]: unhandled `err`

**Comment by dpordomingo** [comment]
I couldn't post before merging because the connectivity issues in the office :(

File: Makefile [hunk 3]
```diff
@@ -96,3 +108,16 @@ toc:
...
 	chmod a+x gh-md-toc
 	./gh-md-toc --insert README.md
+
+.PHONY: ci-start-bblfsh
+ifeq ($(OS),Darwin)
+ci-start-bblfsh:
+	@echo "running bblfsh is unsupported on mac os ci"
+else
+ci-start-bblfsh:
+	docker run -d --name bblfshd --privileged -v $(HOME)/bblfshd:/var/lib/bblfshd -p "9432:9432" bblfsh/bblfshd:v2.9.0
```
[the remark this answers is not in the document]
Comment (smacker) [reply Makefile]: Please explain why.

I don't see any problem with it. Makefile has `ci-install` and `prepare-services` targets already. bblfsh is just one more service.

File: cmd/lookout-sdk/event.go (hunk 2 above)
Comment (smacker) [reply cmd/lookout-sdk/event.go]: thanks! fixed: https://github.com/src-d/lookout/commit/650fcdd90a90f34713bcabc198ca091f04eef458
